[Zhang Yuhao] iP

YH-15 opened the pull request:
# Pull Request
## Some Bullet Points
- First
- Second

## Some Number List
1. First **Again** 
2. Second _Again_

## Fenced code block (with syntax highlighting)
Some sample code:
```ruby
command = in.nextLine().trim();
commandList = command.split(" ", 2);
command = (Parser.multipleVariable(commandList[0])) ? commandList[0] : command;
type = CommandType.map(command);
```
## Task list
- [x] Survive
- [ ] https://github.com/YH-15/ip

## Emoji
@YH-15  💯 Nice!! :grin:

## Blockquote
> ~~This is some random quote.~~

## Hyper link
My repo link [YH-15](https://github.com/YH-15/ip)

## Inline code
This `public static void list(TaskList taskList)` function is aimed to list all the ongoing tasks.

YH-15 committed:
Level-1

YH-15 committed:
Level-2

YH-15 committed:
Level-3

YH-15 committed:
Level-4

YH-15 committed:
Level-5

YH-15 committed:
Level-6

YH-15 committed:
A-TextUiTesting

YH-15 committed:
code style change + OOP style + A-Exceptions

YH-15 committed:
More exception handlings + define Duke Exception class

YH-15 committed:
Fix "delete" feature bug

YH-15 committed:
A-Enums

YH-15 committed:
Adds cache fundtion to Duke, Clears some bugs

YH-15 committed:
Adds date & time functions

YH-15 committed:
Merge branch 'branch-Level-7'

YH-15 committed:
Merges branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/Event.java
#	src/main/java/Interface.java

YH-15 committed:
Adds more OOP more structures

YH-15 committed:
Moves codes with similiar functions to the same packages

YH-15 committed:
Adds Gradle support

YH-15 committed:
Adds JUnit test
Incorporates checkstyle to Gradle

YH-15 committed:
Create JAR file

YH-15 committed:
Adds JavaDoc

YH-15 committed:
Revert "Adds JavaDoc"

This reverts commit 59ee06eae2261df6a194b6b33c53e0941a5419a6.

YH-15 committed:
Revert "Revert "Adds JavaDoc""

This reverts commit 940ecb31ea752e8efcc755eb8a2e17ae0785054f.

YH-15 committed:
Adds more JavaDoc

YH-15 committed:
Add more JavaDoc to satisfy code standard.
Clears some code standard issues.

YH-15 committed:
Adds find function

YH-15 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Ui.java
#	src/main/java/duke/parser/Parser.java

YH-15 committed:
Commits to create a Checkstyle tag, task already completed before.

chengda300 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,171 @@
+        String suggestion;
+        String[] commandList;
+        Integer taskIndex = null;
+        Integer i;
```
This "Integer i" is only used in a "for" loop. Since the variable name is very vague, maybe it might be possible to just declare it directly within the loop to reduce confusion?

YH-15 replied:
Yes, indeed that's a better way to do it thx!

chengda300 commented on src/main/java/duke/dukeexception/IncomplateCommandException.java:
```diff
@@ -0,0 +1,17 @@
+/**
+ * The exception shown when user entered commands that needs (multiple) input, but input not found.
+ */
+public class IncomplateCommandException extends DukeException {
```
Perhaps there is a typo error in the class name?

YH-15 replied:
Thank you for pointing this out! I have made the changes.

chengda300 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,120 @@
+     * @param str The command input by user.
+     * @return A boolean value to indicate if the command needs another input.
+     */
+    public static boolean multipleVariable(String str) {
```
Is it possible to make it sound more like a boolean, like "hasMultipleVariables"?

YH-15 replied:
Thx for your input, I have made changes correspondingly

chengda300 commented on src/main/java/duke/storage/Cache.java:
```diff
@@ -0,0 +1,121 @@
+
+package duke.storage;
```
It is possible that this line has been misplaced to line 2, I think it was supposed to be in line 1?

YH-15 replied:
ohh i did not know this before, thanks for sharing
Have moved it to line 1

chengda300 commented on src/main/java/duke/storage/Cache.java:
```diff
@@ -0,0 +1,121 @@
+     * @return A recovered list of previous work or a blank list if no cache.
+     * @throws DukeException If IOException occurs during the process.
+     */
+    public static TaskList recovery(File f) throws DukeException, FileNotFoundException {
```
recovery sounds more like a noun than a verb, maybe it is possible to make it sound like a verb instead?

YH-15 replied:
Yes, I did not realized this, thanks for pointing out the issue.

chengda300 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,124 @@
+    /**
+     * Update the index of a particular task to be index -1 after removal.
+     */
+    public void updateRemoval() {
```
updateRemoval sounds more like a noun, is it possible to make it sound more like a verb?

chengda300 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,124 @@
+     * Returns whether the task is completed.
+     * @return True if done, False if have not yet done.
+     */
+    public boolean checkDone() {
```
checkDone sounds more like a verb. Perhaps making it sound more like a boolean instead?

YH-15 replied:
Indeed, I have changed to isDone()

chengda300 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,124 @@
+     * @param str the substring for searching.
+     * @return The indicator.
+     */
+    public boolean findName(String str) {
```
findName sounds like a verb. Perhaps make it sound more like a boolean instead (such as "hasName")?

YH-15 replied:
Sure, thx for the reminder, I have updated the method name as hasName()

chengda300 reviewed (commented):
Overall, the code largely follows coding standards, other than a few naming problems (mostly booleans).
I also like how the classes are packaged into different packages appropriately.

chengda300 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,171 @@
+
+            try {
+                switch (type) {
+                case BYE: {
+                    Ui.bye();
+                    lastCommandOrNot = true;
+                    break;
+                }
```
Statement within case blocks do not need curly braces according to the coding standards.
Is it possible to remove them?

YH-15 replied:
Oh thanks for reminder, I have made the changes!

YH-15 committed:
no message

YH-15 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat

YH-15 committed:
Adds CheckStyle

YH-15 committed:
Adds Varargs in find function

YH-15 committed:
Fixes bug

YH-15 committed:
Merge remote-tracking branch 'origin/heads/A-CheckStyle'

YH-15 committed:
Adds GUI, Duke response still pending format change

YH-15 committed:
Adopts PR review suggestions

YH-15 committed:
Completed GUI implementation

* Solved the issue that DialogBox of Duke cannot be displayed properly.
* Redirected output string to a new ByteArrayOutputStream, so that the
response could be recorded by Duke and display in the DialogBox.
* Ensured that both GUI and CLI work fine with current Duke version.

Debuged "find" command

Previously, it cannot take multiple keywords at a time, this issue was
solved after debugging.

YH-15 committed:
Merge pull request #1 from YH-15/branch-Code-Style

Completed GUI implementation

YH-15 committed:
Adds assertion checks

* checked if the output stream is alr captured before the use of
OutputRedirector class
* checked if the duke.txt file was successfully created by
Files.createDirectories(path)

YH-15 committed:
Deletes a residual class.

YH-15 committed:
(residual comment for commit, my branch naming not aligned with what
is required)

YH-15 committed:
Merge pull request #2 from YH-15/branch-A-Assertions

merge A-Assertions

YH-15 committed:
no message

YH-15 committed:
Updates some JavaDoc

YH-15 committed:
Merge pull request #3 from YH-15/branch-A-CodeQuality

Updates some JavaDoc

YH-15 committed:
Merge pull request #4 from YH-15/branch-A-Assertions

no message

YH-15 committed:
C - Undo

YH-15 committed:
Improved GUI
Added UG

YH-15 committed:
Updated User Guide

YH-15 committed:
no message

YH-15 committed:
Updated Release

YH-15 committed:
Removed the javafx block from build.gradle file

YH-15 committed:
Add User Guide
Fix error handling

YH-15 committed:
Fix Checkstyle

YH-15 committed:
Fix some more bugs
